### ISSUE TITLE
With [prevAll] and then with [wrapAll], the order of the nodes in the dom changes

### ISSUE THREAD

**Issue by lovecodeyou** [jquery/jquery#5149]
Bug Reports:
With [prevAll] and then with [wrapAll], the order of the nodes in the dom changes

### Description ###
```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
    <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.1/dist/jquery.min.js"></script>
</head>
<body>
    <div id="row" style="border: 1px solid black;">
        <span>【one】</span>
        <span>【two】</span>
        <span id="middle">【middle】</span>
        <span>【three】</span>
        <span>【four】</span>
    </div>


    <div id="jqueryChange" style="border: 1px solid black;margin-top: 50px;">
        <span>【one】</span>
        <span>【two】</span>
        <span id="middle">【middle】</span>
        <span>【three】</span>
        <span>【four】</span>
    </div>    
</body>

<script>

    $(function() {

        $("#jqueryChange #middle").each(function () {
                    $(this).prevAll()
                        .wrapAll("<p/>")
                        .end()
                        .nextAll()
                        .wrapAll("<p/>")
        })

    })

</script>

</html>
```

![image](https://user-images.githubusercontent.com/38997988/198357744-9e0985d5-8b65-4d7d-ac0b-1f5fc6b389ef.png)

**Comment by mgol** [comment]
Thanks for the report.

This is an interesting edge case. Generally, jQuery objects are collections sorted along the document order. However, `.prevAll()` breaks this invariant as it contains the elements in the backwards order. Various manipulation functions are largely inaware of this peculiarity so our DOM manipulation takes elements one by one and applies the relevant callback. In the context of `.append()` which is used internally via `.wrapAll()` it appends elements one by one, changing their order.

We'll discuss how to handle this.

**Comment by timmywil** [comment]
We discussed this in the meeting. Rather than trying to predict desired behavior in all manip methods, we're going to plan on adding `.uniqueSort()` as an instance method (it already exists on the jQuery object at `jQuery.uniqueSort`) and improve the docs for `.prevAll()` and `.parents()` to be clear about the implications of getting a set in reverse order. So, the solution here would become `.prevAll().uniqueSort().wrapAll('<p/>')`.

**Comment by mgol** [comment]
I created issue #5166 to be more explicit about `uniqueSort`.

**Comment by mgol** [comment]
The `.unique()` method has been implemented in #5168, it will be included in jQuery `3.7.0`. Docs updates mentioning the use of `jQuery.uniqueSort()` have been added in https://github.com/jquery/api.jquery.com/pull/1215 and are live at https://api.jquery.com/prevAll/. A draft PR switching that page to use the `.uniqueSort()` method is submitted at https://github.com/jquery/api.jquery.com/pull/1216 and will be merged after jQuery `3.7.0` gets released.

Since we don't plan to implement the original proposal from this issue, I'm going to remove the milestone. However, all of the ways we decided to address it have been implemented so I think we can close it.
